spinn_front_end_common/interface/interface_functions/application_runner.py

```diff
@@ -103,15 +89,43 @@ def __call__(
                 time.sleep(time_to_wait)
                 timeout = time_threshold
             else:
-                logger.info(
-                    "Application started - waiting until finished")
+                logger.info("Application started - waiting until finished")
 
-            txrx.wait_for_cores_to_be_in_state(
-                executable_targets.all_core_subsets, app_id,
-                [CPUState.FINISHED, CPUState.PAUSED], timeout=timeout)
+            for executable_type in executable_types:
+                txrx.wait_for_cores_to_be_in_state(
+                    executable_types[executable_type], app_id,
+                    executable_type.end_state, timeout=timeout)
 
         if (notification_interface is not None and
                 send_stop_notification and runtime is not None):
             notification_interface.send_stop_pause_notification()
 
         return True, no_sync_changes
+
+    @staticmethod
+    def _determine_simulation_sync_signals(executable_types, no_sync_changes):
+        """ sorts out start states, and creates core subsets of the states for
+        further checks.
+
+        :param no_sync_changes: sync counter
+        :param executable_types: the types of executables
+        :return: the sync signal and updated no_sync_changes
+        """
+        sync_signals = list()
+
+        if ExecutableType.USES_SIMULATION_INTERFACE in executable_types:
+            if no_sync_changes % 2 == 0:
+                sync_signals.append(Signal.SYNC0)
+            else:
+                sync_signals.append(Signal.SYNC1)
+            # when it falls out of the running, it'll be in a next sync \
+            # state, thus update needed
+            no_sync_changes += 1
+
+        # handle the sync states, but only send once if they work with \
+        # the simulation interface requirement
+        if ExecutableType.SYNC in executable_types:
+            if Signal.SYNC0 not in sync_signals:
+                sync_signals.append(Signal.SYNC0)
+
+        return sync_signals, no_sync_changes
```

Comment:
Doesn't this mean you can have multiple SYNC signals being sent - isn't that dangerous?  Maybe this can also be added to the ExecutableType

Reply:
no way around it from what i can see. even putting it into the executable type has issue, as simulation changes its sync state based off reset.

Reply:
OK, so this state should never occur, because the SIMULATION binaries will go into SYNC0 in the first case anyway, thus all is well.  Maybe this should be documented though...

Reply:
and its never multple of the same signal. jsut possible both sync0 and sync1

Reply:
So you should never send both sync0 and sync1 to the machine at the same time, and this code looks like it can, by adding sync1 from the SIMULATION code and SYNC0 from the SYNC code.  Thus if it ever gets to the point where it might, it should error.  That point should never occur in code from what I can see, but then who knows what someone might try in the future...

Reply:
why cant you? if they in different sync states, shouldnt sending both get them all to start? whats the limit here?

Reply:
fixed